explorer/interpreter/expression_checker.cpp

```diff
@@ -0,0 +1,1175 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
Is this supposed to be added in this PR?

Reply:
> Is this supposed to be added in this PR?

Whoops, fixed